edge/pkg/metamanager/client/pod.go

```diff
@@ -206,3 +230,47 @@ func handleMqttMeta() (*corev1.Pod, error) {
 	}
 	return &pod, nil
 }
+
+func patchOnlyEdgePod(name, namespace string, patchBytes []byte) error {
+	url := "/api/v1/namespaces/" + namespace + "/pods/" + name
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return err
+	}
+	resolver := &request.RequestInfoFactory{
+		APIPrefixes:          sets.NewString("api", "apis"),
+		GrouplessAPIPrefixes: sets.NewString("api"),
+	}
+	apiRequestInfo, err := resolver.NewRequestInfo(req)
+	ctx := request.WithRequestInfo(context.TODO(), apiRequestInfo)
+	key, err := metaserver.KeyFuncReq(ctx, "")
+	results, err := v2.RawMetaByGVRNN(metaserver.ParseKey(key))
+	if err != nil {
+		return err
+	}
+	if len(*results) == 0 {
+		return fmt.Errorf("not found pod:%s/%s", namespace, name)
+	}
+	originalPod := &corev1.Pod{}
+	err = json.Unmarshal([]byte((*results)[0].Value), originalPod)
+	scheme := runtime.NewScheme()
+	GV := schema.GroupVersion{Group: GroupName, Version: "v1"}
+	scheme.AddKnownTypes(GV, &corev1.Pod{})
+	defaulter := runtime.ObjectDefaulter(scheme)
+	updatedPod := &corev1.Pod{}
+	err = util.StrategicPatchObject(context.TODO(), defaulter, originalPod,
+		patchBytes, updatedPod, &corev1.Pod{}, "")
+	if err != nil {
+		return err
+	}
+	if !reflect.DeepEqual(originalPod, updatedPod) {
+		podContent, err := json.Marshal(updatedPod)
+		if err != nil {
+			return err
+		}
+		resource := fmt.Sprintf("%s/%s/%s", updatedPod.Namespace, model.ResourceTypePod, name)
+		msg := message.BuildMsg(modules.MetaGroup, "", modules.EdgedModuleName, resource, model.UpdateOperation, podContent)
+		imitator.DefaultV2Client.Inject(*msg)
```

Comment:
Is it possibe to use the `client-go` to patch the status from metaserver instead of performing database directly here?

Reply:
it's ok